onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalsListFragment.java

```diff
@@ -268,6 +269,23 @@ public View onCreateView(LayoutInflater inflater,
     }
 
     @Override
+    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+        RouteFavoriteDialogFragment dialogFragment = (RouteFavoriteDialogFragment)
+                getFragmentManager().findFragmentByTag(RouteFavoriteDialogFragment.TAG);
+        if (dialogFragment != null) {
+            dialogFragment.setCallback(new RouteFavoriteDialogFragment.Callback() {
+                @Override
+                public void onSelectionComplete(boolean savedFavorite) {
+                    if (savedFavorite) {
+                        refreshLocal();
+                    }
+                }
+            });
+        }
+        super.onViewStateRestored(savedInstanceState);
```

Comment:
I did a little bit google search on it. Seems most people prefer to call first the super class then run their code. However there is no official recommendation exist for this.

PS: Example google fragment code calls the super class first. http://developer.android.com/reference/android/app/Fragment.html

Therefore, I am going to move `super.onViewStateRestored(savedInstanceState);` to the beginning.